test/params_api_test.c

```diff
@@ -501,13 +505,29 @@ static int test_param_construct(void)
     params[n++] = OSSL_PARAM_construct_octet_ptr("octptr", &vp, 0);
     params[n] = OSSL_PARAM_construct_end();
 
+    switch(tstid) {
```

Comment:
Have I ever mentioned that I hate this test methodology??? :(

Reply:
I can create separate tests to do the same thing that just call a function with different arguments (not sure this makes much of a difference)

Reply:
Don't worry about it, it's what is used through the rest of the tests.
I'd prefer an array of structs defining things -- it's much easier to add/remove cases.